### PR TITLE
Fix OCS Share API path response

### LINKED ISSUE #24997: Shared from you view shows wrong folder names
When I share a folder FOLDER-A to a person and he is first renaming to FOLDER-B and then sharing it as well, in "Shared from you" (or similiar in english) view shows still the olf name FOLDER-A. In the folder view he will the the new folder name.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #24997

There are only 3 possible types of users that can request a share via the OCS Share API:
- owner
- inituator
- recipient

Else they won't find the share.
This makes sure we get the share for the current user. So that the path is always something meaningfull for that user.

CC: @nickvergessen @PVince81 @icewind1991 @scriptPilot

### CODE REVIEW

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 1]
```diff
@@ -123,8 +116,10 @@ protected function formatShare(\OCP\Share\IShare $share) {
 			'displayname_file_owner' => $shareOwner !== null ? $shareOwner->getDisplayName() : $share->getShareOwner(),
 		];
 
-		$node = $share->getNode();
-		$result['path'] = $this->rootFolder->getUserFolder($localUser)->getRelativePath($node->getPath());
+		$userFolder = $this->rootFolder->getUserFolder($this->currentUser->getUID());
+		$node = $userFolder->getById($share->getNodeId())[0];
```
Comment (nickvergessen) [inline apps/files_sharing/lib/API/Share20OCS.php]: check for non empty result from `getById` ?

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 2]
```diff
@@ -100,15 +100,8 @@ public function __construct(
...
-		// for federated shares the owner can be a remote user, in this
-		// case we use the initiator
-		if ($this->userManager->userExists($share->getShareOwner())) {
-			$shareOwner = $this->userManager->get($share->getShareOwner());
-			$localUser = $share->getShareOwner();
-		} else {
-			$shareOwner = $this->userManager->get($share->getSharedBy());
-			$localUser = $share->getSharedBy();
-		}
+		$shareOwner = $this->userManager->get($share->getShareOwner());
```
Comment (nickvergessen) [inline apps/files_sharing/lib/API/Share20OCS.php]: what about the federated share case where the owner is not local, `$shareOwner` is null then

Comment (nickvergessen) [inline apps/files_sharing/lib/API/Share20OCS.php]: Ah, just saw that there is handling for null already in place

File: apps/files_sharing/lib/API/Share20OCS.php (hunk 1 above)
Comment (rullzer) [inline apps/files_sharing/lib/API/Share20OCS.php]: Ah right... in case of messed up shares... yeah lets check that...

**Comment by rullzer** [comment]
This basically obsoletes the 'target' in the OCS SHare API response.

**Comment by PVince81** [comment]
Tested, works :+1: 

@rullzer also observed on 9.0, please backport after review

CC @dragotin @DeepDiver1975

File: apps/files_sharing/lib/API/Share20OCS.php (hunk 2 above)
Comment (PVince81) [inline apps/files_sharing/lib/API/Share20OCS.php]: So is this PR acceptable ?

**Comment by PVince81** [comment]
Need seconds reviewer @ChristophWurst @nickvergessen @DeepDiver1975 @owncloud/sharing

**Comment by icewind1991** [comment]
:+1: looks good

**Comment by PVince81** [comment]
@rullzer please backport to 9.0

**Comment by rullzer** [comment]
Backport in #25067

**Pull request merged**